Lesson10/src/homework10/Buyer.java

```diff
@@ -16,20 +16,22 @@ public Buyer(String name, List<CashRegister> cashRegisters, String currency) {
 
     @Override
     public void run() {
-        choiceCashRegister().serveCustomer(this);
+        try {
+            choiceCashRegister().serveCustomer(this);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
     }
 
-    private CashRegister choiceCashRegister() {
-        Random random = new Random();
+    private CashRegister choiceCashRegister() throws Exception {
         List<CashRegister> similarCashRegister = new LinkedList<>();
         for (CashRegister i : cashRegisters) {
             if (i.getCurrency().equals(currency)) {
                 similarCashRegister.add(i);
             }
         }
         if (similarCashRegister.size() == 0) {
-            System.out.println(getName() + "\tWe don't use this currency,  You could use our ATM");
-            return null;
+            throw new Exception(getName() + "\tWe don't use this currency,  You could use our ATM");
```

Comment:
please don't throw Exception. Create your own one and throw it. Throwing and catching Exception is a bad practice, and it affects performance